src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -0,0 +1,261 @@
+        writer.close()
+        connection.connect()
+        connection
+    }
+
+    /**
+     * Generate ElasticSearch query to list locations by campus, type and full text search.
+     *
+     * @param trimmedCampus
+     * @param trimmedType
+     * @param trimmedQ
+     * @param pageNumber
+     * @param pageSize
+     * @param query
+     * @return
+     */
+    private def getESSearchQuery(String trimmedCampus, String trimmedType, String trimmedQ, int pageNumber,
+                                 int pageSize, def query) {
+        def esQuery = [
+                "query": [
+                    "filtered": [
+                        "filter": [
+                            "bool": [
+                                "must": []
+                            ]
+                        ],
+                        "query" : [:]
+                    ]
+                ],
+                "from" : (pageNumber - 1) * pageSize,
+                "size" : pageSize
+        ]
+
+        if (trimmedCampus) {
+            esQuery.query.filtered.filter.bool.must += ["term": ["attributes.campus": trimmedCampus]]
+        }
+
+        if (trimmedType) {
+            esQuery.query.filtered.filter.bool.must += ["term": ["attributes.type": trimmedType]]
+        }
+
+        if (trimmedQ) {
+            query = ["query_string": ["query": trimmedQ]]
+        }
+
+        esQuery.query.filtered.query = query
+        esQuery
+    }
+
+    /**
+     * Sanitizes the search query string by replacing illegal characters with spaces.
+     *
+     * @param searchQuery
+     * @return sanitized search query
+     */
+    private static String sanitize(String searchQuery) {
+        if (!searchQuery) {
+            return ""
+        }
+
+        illegalCharacterPattern.matcher(searchQuery).replaceAll(' ')
+    }
+
+    /**
+     * Returns the value for an array parameter in the GET string.
+     *
+     * The JSONAPI format reserves the page parameter for pagination. This API uses page[size] and page[number].
+     * This function allows us to get just value for a specific parameter in an array.
+     *
+     * @param key
+     * @param index
+     * @param queryParameters
+     * @return
+     */
+    public static String getArrayParameter(String key, String index,  MultivaluedMap<String, String> queryParameters) {
+        // @todo: this function should probably be moved to the skeleton
+        for (Map.Entry<String, List<String>> entry : queryParameters.entrySet()) {
+            // not an array parameter
+            if (!entry.key.contains("[") && !entry.key.contains("]")) {
+                continue
+            }
+
+            int a = entry.key.indexOf('[')
+            int b = entry.key.indexOf(']')
+
+            if (entry.key.substring(0, a).equals(key)) {
+                if (entry.key.substring(a + 1, b).equals(index)) {
+                    return entry.value?.get(0)
+                }
+            }
+        }
+
+        null
+    }
```

Comment:
it might be worth implementing this using regular expressions. something like `\[(.*)\]` would enable you to make use of the parenthesized group.

also consider whether this method should be public.